### ISSUE TITLE
Listbox with label isn't read properly by VoiceOver

### ISSUE THREAD

**Issue by robsquires** [tailwindlabs/headlessui#2817]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

For example: v1.7.17

**What browser are you using?**

Chrome, Safari, Firefox, MacOSX

**Reproduction URL**

Github repo: https://github.com/robsquires/polished-cookies-mqyxtz
Codesandbox: https://codesandbox.io/p/github/robsquires/polished-cookies-mqyxtz/main

**Describe your issue**
I'm seeing an issue with the Listbox component not being read correctly by Voiceover. I believe the correct behaviour is that when the listbox is opened, Voiceover should read the text _{Text for Option 1}, not selected, (1 of {X})_. Eg:

<img width="661" alt="Screenshot 2023-10-27 at 16 23 01" src="https://github.com/tailwindlabs/headlessui/assets/616321/1b8255ff-3921-4e14-8531-4680e21d31aa">

This can be observed when using the ListBox in single selection mode - `<Listbox multiple={false} />`.

This issue I'm seeing is when the Listbox is used in multiple selection mode - `<Listbox multiple />` *and* used in-conjuntion with a `Listbox.Label`. In this case Voiceover reads the text from the label, instead the details outlined previously:

<img width="674" alt="Screenshot 2023-10-27 at 16 29 16" src="https://github.com/tailwindlabs/headlessui/assets/616321/2c41ce34-050c-4c09-be9c-435e1a2c97cd">

Once an item is selected and the listbox is re-opened, VoiceOver reads the options correctly.

The codesandbox outlines which combinations are exhibiting issues:

- Listbox single selection, no label ✅
- Listbox single selection, label ✅
- Listbox multiple selection, no label ✅
- Listbox multiple selection, label ❌

I've done some digging and it looks to be a combination of `aria-labelledby` and `aria-activedescendant` which are applied differently depending on how the component is configured.

| aria-labelledby  | aria-activedescendant | Outcome | 
| - | - | - |
| not present | present | ✅ |
| not present | not present | ✅ |
| present | present | ✅ |
| present | not present | ❌ |

When the Listbox is in multiple mode it doesn't look like `aria-activedescendant` unless an item is selected, which is why the addition of a label breaks things.

There's a v. similar issue raised for the comobox https://github.com/tailwindlabs/headlessui/issues/2736

Feels like this is a bug but I'd be happy to know if I'm configuring the component wrong too!

Thanks!

**Comment by RobinMalfait** [comment]
Hey!

There is currently a bug in Chrome + VoiceOver for this use case. I filed a bug report for this. Luckily there was a workaround we could apply in Headless UI itself. (More info about that: #2824).

This should be fixed by #2824, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.
